src/components/Layout/Layout.jsx

```diff
@@ -1,7 +1,35 @@
 import "./Layout.scss";
+import Menu from "../Menu/Menu";
+import MenuIcon from "../MenuIcon/MenuIcon";
+import { useState } from "react";
 
-const Layout = ({ children }) => {
-  return <div className="layout">{children}</div>;
-};
+const Layout = ({ children, isWithMenu }) => {
+  if (!isWithMenu) {
+    return <div className="layout">{children}</div>;
+  }
+
+  //Layout with Menu Icon
+  const [showMenu, setShowMenu] = useState(false);
 
+  const toggleMenu = () => {
+    setShowMenu(!showMenu);
+  };
+
+  return (
+    <>
+      <div className="layout">
+        {showMenu && (
+          <>
+            <div className="menu__shadow"></div>
```

Comment:
all the classes in this file should have a block of `layout`, so could this be changed to `layout__menu-shadow` ? or `layout__shadow`